### PR TITLE
add ultanalyse for supporting Apollo

### PR DESCRIPTION
**Pull request by HanochZhong**

### CODE REVIEW

**Comment by jinghaomiao** [comment]
@HanochZhong  Please add unit test with your pull request. Thanks!

File: modules/common/adapters/BUILD [hunk 1]
```diff
@@ -103,6 +103,7 @@ cc_library(
         "//modules/planning/proto:planning_proto",
         "//modules/prediction/proto:prediction_proto",
         "//modules/routing/proto:routing_proto",
+        "//modules/drivers/ultanalyse/proto:ultanalyse_proto",
```
Comment (xiaoxq) [inline modules/common/adapters/BUILD]: Please make rows sorted.

File: modules/common/adapters/adapter.h [hunk 2]
```diff
@@ -46,6 +46,7 @@
 #include "sensor_msgs/PointCloud2.h"
 #include "std_msgs/String.h"
 #include "velodyne_msgs/VelodyneScanUnified.h"
+#include "std_msgs/Int32MultiArray.h"
```
Comment (xiaoxq) [inline modules/common/adapters/adapter.h]: ditto, as well as all following places.

File: modules/common/adapters/adapter.h [hunk 3]
```diff
@@ -401,6 +402,11 @@ class Adapter : public AdapterBase {
                 IdentifierType<velodyne_msgs::VelodyneScanUnified>) {
     return false;
   }
+  bool FeedFile(const std::string& message_file,
```
Comment (xiaoxq) [inline modules/common/adapters/adapter.h]: Please leave blank line between functions.

File: modules/drivers/ultanalyse/ultanalyse.h [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file
+ */
+
+#ifndef MODEULES_ULTANALYSE_ULTANALYSE_H_
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/ultanalyse.h]: Use #pragma once.

File: modules/drivers/ultanalyse/common/ultanalyse_gflags.cc [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/common/ultanalyse_gflags.cc]: 2019. Please also fix others.

File: modules/drivers/ultanalyse/common/ultanalyse_gflags.cc [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/drivers/ultanalyse/common/ultanalyse_gflags.h"
+
+DEFINE_string(node_name, "ultanalyse", "The chassis module name in proto");
+DEFINE_string(module_name, "ultanalyse", "Module name");
+DEFINE_string(adapter_config_filename,
+             "modules/ultanalyse/conf/ultanalyse.conf",
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/common/ultanalyse_gflags.cc]: Fix indents. You could try this tool: ./scripts/clang-format.sh

File: modules/drivers/ultanalyse/main.cc [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/******************************************************************************
+ * Copyright 2018
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/main.cc]: Fix copyright, as well as other places.

File: modules/drivers/ultanalyse/ultanalyse.cc [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+ *****************************************************************************/
+#include "modules/drivers/ultanalyse/ultanalyse.h"
+
+#include <cmath>
+#include <vector>
+#include "modules/common/adapters/adapter_gflags.h"
+#include "modules/common/adapters/adapter_manager.h"
+#include "modules/common/log.h"
+#include "modules/drivers/ultanalyse/common/ultanalyse_gflags.h"
+#include "ros/include/ros/ros.h"
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/ultanalyse.cc]: Put third party headers before Apollo modules.

File: modules/drivers/ultanalyse/ultanalyse.cc [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+      obst_analyse_.add_obstinfo();
+      obst_analyse_.mutable_obstinfo(count)->set_location_y(obst_location_y);
+      obst_analyse_.mutable_obstinfo(count)->set_location_x(obst_location_x);
+      obst_analyse_.mutable_obstinfo(count)->set_detec_sonar(i+1);
+      obst_analyse_.mutable_obstinfo(count)->set_obst_distance(obst_distance);
+      count++;  // count the number of the detected obstacles
+
+
+
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/ultanalyse.cc]: Remove extra blank lines.

File: modules/drivers/ultanalyse/ultanalyse.h [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+#define MODEULES_ULTANALYSE_ULTANALYSE_H_
+
+#include <map>
+#include <mutex>
+#include <queue>
+#include <string>
+
+// #include "modules/canbus/proto/chassis.pb.h"
```
Comment (xiaoxq) [inline modules/drivers/ultanalyse/ultanalyse.h]: Remove all unused code.

File: modules/common/adapters/adapter_gflags.cc [hunk 11]
```diff
@@ -1,5 +1,5 @@
 /******************************************************************************
- * Copyright 2017 The Apollo Authors. All Rights Reserved.
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline modules/common/adapters/adapter_gflags.cc]: No need to change year of existing files.

**Comment by xiaoxq** [comment]
It's not good practice to guess and try fixing one-file-by-one-file on the web. It costs a lot of CI resources. Please setup environment at local, develop, run `apollo.sh build`, `apollo.sh lint`, `apollo.sh test` to make sure it's all clear before making commit. Thanks!

**Comment by jinghaomiao** [comment]
@HanochZhong I am closing this pull request as it is pending for over 2 months. Please make changes according to the review and resubmit the request if necessary. Thanks for supporting Apollo.

**Pull request closed without merging**